core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/commandqueue/handlers/DeprovisionCommandHandler.java

```diff
@@ -0,0 +1,42 @@
+package org.eclipse.dataspaceconnector.transfer.core.commandqueue.handlers;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.transfer.command.CommandHandler;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.commands.DeprovisionRequest;
+
+import static java.lang.String.format;
+
+public class DeprovisionCommandHandler implements CommandHandler<DeprovisionRequest> {
+
+    @Inject
+    private TransferProcessStore store;
+
+    public DeprovisionCommandHandler() {
+        // used for reflection + dependency injection
+    }
+
+    public DeprovisionCommandHandler(TransferProcessStore store) {
+        this.store = store;
+    }
+
+    @Override
+    public void handle(DeprovisionRequest command) {
```

Comment:
Should this command be launched also by the `deprovision` method on the TPM? Because at the moment there's some duplicated code there

Reply:
Those methods are now marked to be removed and so they should be. I'll replace them with adding commands.